src/main/java/com/teragrep/jos_01/procfs/status/process/Statm.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Java Operating System Statistics JOS-01
+ * Copyright (C) 2021-2024 Suomen Kanuuna Oy
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ *
+ * Additional permission under GNU Affero General Public License version 3
+ * section 7
+ *
+ * If you modify this Program, or any covered work, by linking or combining it
+ * with other code, such other code is not for that reason alone subject to any
+ * of the requirements of the GNU Affero GPL version 3 as long as this Program
+ * is the same Program as licensed from Suomen Kanuuna Oy without any additional
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.jos_01.procfs.status.process;
+
+import com.teragrep.jos_01.procfs.status.CharacterDelimited;
+import com.teragrep.jos_01.procfs.status.Text;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Instant;
+import java.util.*;
+
+// Provides information about memory usage of the process.
+// Fields are on a single row, separated by spacebars. Contains 7 fields.
+// All fields are integers.
+public class Statm implements Text {
+
+    private final Logger LOGGER = LoggerFactory.getLogger(Statm.class);
+    private final Instant timestamp;
+    private final long size;
+    private final long resident;
+    private final long shared;
+    private final long text;
+    private final long lib;
+    private final long data;
```

Comment:
May we know why this class is named data and what they should tell about memory to us?

Reply:
It's the memory usage of the processes' data and stack, in pages.
I chose these names because  they are used in the man pages for this proc file:
https://man7.org/linux/man-pages/man5/proc_pid_statm.5.html